crates/sui/src/upgrade_compatibility.rs

```diff
@@ -592,14 +601,26 @@ fn compare_packages(
 
     // use colors but inline
     Err(anyhow!(
-        "{}\nUpgrade failed, this package requires changes to be compatible with the existing package. \
+        "{}{}\nUpgrade failed, this package requires changes to be compatible with the existing package. \
         Its upgrade policy is set to 'Compatible'.",
```

Comment:
Before we turn this on for good, do we want to tweak this message? What do we do locally (in the compatibility check) if the upgrade policy is not 'Compatible'?

Reply:
This is something I glossed over, I'll do a bit of the work here to stub the main `to_diag` function part out and fix this message it out it will require its own PR I think (besides the simple formatting)